test/TestProjects/Customizations-Typespec/src/Customizations/ModelToAddAdditionalSerializableProperty.cs

```diff
@@ -0,0 +1,37 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#nullable disable
+
+using System.Text.Json;
+using Azure.Core;
+
+namespace CustomizationsInCadl.Models
+{
+    /// <summary> Model to add additional serializable property. </summary>
+    public partial class ModelToAddAdditionalSerializableProperty
+    {
+        /// <summary> New property to serialize. </summary>
+        [CodeGenMemberSerialization("additionalSerializableProperty")]
+        public int AdditionalSerializableProperty { get; set; }
+
+        /// <summary>
+        /// Required int.
+        /// This property is mocking this scenario:
+        /// In the SDK, this property is defined as int, but in the actual traffic, this property comes as a string.
+        /// We use this attribute to fix its serialization and deserialization using the following two methods
+        /// </summary>
+        [CodeGenMemberSerializationHooks(nameof(WriteRequiredInt), nameof(DeserializeRequiredInt))]
+        public int RequiredInt { get; set; }
+
+        private void WriteRequiredInt(Utf8JsonWriter writer)
+        {
+            writer.WriteStringValue(RequiredInt.ToString());
```

Comment:
We should use aggressive inlining on these methods but since they are hand written we should add an analyzer that ensures this.  Please create a follow up work item in azure-sdk-tools to add this analyzer.